### PR TITLE
Rework

### LINKED ISSUE #7: TPR name unclear
I'm very excited about this operator and giving it a try. In https://github.com/pires/nats-operator/blob/master/pkg/controller/controller.go#L38 the TPR name is set to `management.nats.io` which also seems to be the TPR that's created when I run the operator.

The example however defines the kind as `NatsCluster` https://github.com/pires/nats-operator/blob/master/example/example-nats-cluster.yaml

However there is no `natscluster` TRP.

What am I missing?

### LINKED ISSUE #10: Upgrade to latest NATS 1.x.

### LINKED ISSUE #11: Move from TPR to CRD
Read https://coreos.com/blog/custom-resource-kubernetes-v17.

### LINKED ISSUE #13: Move from glide to dep
See https://github.com/golang/dep

### PR DESCRIPTION
**Pull request by pires**
Fixes #7
Fixes #10 
Fixes #11
Fixes #13
Fixes #14 

- [x] [Move from glide to dep](#13)
- [x] Move to versioned [`client-go`](https://github.com/kubernetes/client-go).
- [x] Ensure `dep ensure` runs and there's no need for versioning vendored code.
- [x] Make sure Docker container image is built successfully.
- [x] Replace official image with `quay.io/pires/docker-nats` which includes `route_checker`.

### CODE REVIEW

File: test/e2e/basic_test.go [hunk 1]
```diff
@@ -108,21 +108,21 @@ func TestNatsUpgrade(t *testing.T) {
...
 	if err != nil {
 		t.Fatalf("fail to update cluster version: %v", err)
 	}
 
 	// We have seen in k8s 1.7.1 env it took 35s for the pod to restart with the new image.
-	err = e2eutil.WaitSizeAndVersionReached(t, f.KubeClient, "v1.0.2", 3, 10, testNats)
+	err = e2eutil.WaitSizeAndVersionReached(t, f.KubeClient, "1.0.2", 3, 10, testNats)
 	if err != nil {
 		t.Fatalf("failed to wait new version NATS cluster: %v", err)
 	}
```
Comment (pires) [inline test/e2e/basic_test.go]: Weird, versions are tagged with prefix `v`, see https://github.com/nats-io/gnatsd/tags.

Comment (brunomcustodio) [inline test/e2e/basic_test.go]: It is, indeed, but in Docker Hub (https://hub.docker.com/r/library/nats/tags/) they have no prefix so I ended up removing it. 🤔

Comment (pires) [inline test/e2e/basic_test.go]: Ah, makes total sense.

**Comment by pires** [comment]
Is this ready for rebase?

**Comment by brunomcustodio** [comment]
@pires I believe it is, but please note the latest commit's message:

> Also temporarily commenting 'TestNatsUpgrade' as there are no compatible versions of 'quay.io/pires/docker-nats' below 1.0.2.

If this is not a major issue, then please go straight ahead.

**Pull request merged**